### PR TITLE
F oauth2

### PR DESCRIPTION
**Pull request by Harleymckee**
Incorporate https://github.com/danschultzer/phoenix_oauth2_provider to add Oauth2 endpoints with grant_type=password flow for mobile apps.

Totally open to serving this somewhere besides revelry_phoenix_app_template.

Also @bryanjos I replaced the test of the old api token flow, Let me know if you think we should keep that code and keep two api boundaries. 

client side counter part: https://github.com/revelrylabs/react-native-template-revelry/pull/30

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -75,7 +75,8 @@ defmodule AppTemplate.Mixfile do
       {:prometheus_plugs, "~> 1.1"},
       {:prometheus_process_collector, "~> 1.4"},
       {:libcluster, "~> 3.0"},
-      {:pow, "~> 1.0"}
+      {:pow, "~> 1.0"},
+      {:phoenix_oauth2_provider, "~> 0.5.1", path: "deps/phoenix_oauth2_provider"}
```
Comment (bryanjos) [inline mix.exs]: Is there a reason the path is here?

**Comment by oohnoitz** [comment]
Yeah, agree with @bryanjos it might be good for us to wrap this up into a package we can then add when needed with few lines to configure it. This way its a bit easier to plug into an existing application with some configs and upgrade things without having to duplicate the work across multiple applications.

File: mix.exs (hunk 1 above)
Comment (Harleymckee) [reply mix.exs]: nope that should be deleted

**Comment by Harleymckee** [comment]
Thanks @bryanjos @oohnoitz Good call. I'll follow how Adminable (that phoenix_oauth2_provider has some similar 'meta routing' also) is implemented and reach out if I get stuck.

File: lib/app_template/oauth_password_grant.ex [hunk 2]
```diff
@@ -0,0 +1,23 @@
+defmodule AppTemplate.OAuthPasswordGrant do
+  alias AppTemplate.{User, Repo}
+
+  def authenticate(username, password) do
+    User
+    |> Repo.get_by(email: String.downcase(username))
```
Comment (grossvogel) [inline lib/app_template/oauth_password_grant.ex:6]: NOTE: This will only match if the email is stored in all lowercase in the DB, which I don't think is something we force. Assuming we don't force it, we can instead do something like this (or a more elegant version of it...)
```
    lower_username = String.downcase(username)

    query =
      from(
        u in User,
        where: fragment("lower(?)", u.email) == ^lower_username
      )

    query
    |> Repo.one()
    |> verify_password(password)
```
